woocommerce/woocommerce
# Product Editor Onboarding: Show spotlight for first time visitors

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Show a spotlight and a guided tour for first time visitors, using TourKit and Guide components.


Bug with Guide component: the 'X' button from the modal keeps focusing when navigating between pages.
Limitation with Guide component: it's not possible to know if the Guide was closed or finished, because the same callback is called for both, with no parameters.


Closes #38516 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Enable the block product editor in WooCommerce > Settings > Advanced > Features.
2. Go to Product > Add New.
3. You should see the tour kit step in the bottom left of the editor.
4. The `wcadmin_block_product_editor_spotlight_view` tracks event should be fired.
5. Click the X button.
6. The `wcadmin_block_product_editor_spotlight_dismissed` tracks event should be fired.
7. Refresh the page.
8. You should not see the tour kit step anymore.
9. Delete the option `woocommerce_block_product_tour_shown` using WCA test helper.
10. Refresh the page.
11. The tour kit step should show up again.
12. Click 'View highlights'.
13. You should see the Guide component and be able to navigate between the pages.
14. Navigate until the last page and press 'Close'
15. The `wcadmin_block_product_editor_spotlight_completed` tracks event should be fired.
